geometry-processor.cpp

```diff
@@ -75,11 +75,13 @@ way_helper::~way_helper()
 }
 size_t way_helper::set(const osmid_t *node_ids, size_t node_count, const middle_query_t *mid)
 {
-    //if we don't have enough space already get more
-    if(node_cache.size() < node_count)
+    // other parts of the code assume that the node cache is the size of the way
+    // TODO: Fix this, and use std::vector everywhere
+    if(node_cache.size() != node_count)
         node_cache.resize(node_count);
-    //get the node data
+    // get the node data
     mid->nodes_get_list(&node_cache.front(), node_ids, node_count);
```

Comment:
How's this for a test: First way in file is nodes 1,2,3; second is 4,5; third is 6,7,8 with node 7 missing. I think that catches both.